src/vulkan-renderer/wrapper/texture.cpp

```diff
@@ -39,24 +39,34 @@ Texture::Texture(const wrapper::Device &device, const VkPhysicalDevice graphics_
 
     // Load the texture file using stb_image library.
     // Force stb_image to load an alpha channel as well.
-    stbi_uc *texture_data =
-        stbi_load(file_name.c_str(), &m_texture_width, &m_texture_height, &m_texture_channels, STBI_rgb_alpha);
+    stbi_uc *texture_data = stbi_load(file_name.c_str(), &m_texture_width, &m_texture_height, nullptr, STBI_rgb_alpha);
 
     if (texture_data == nullptr) {
-        throw std::runtime_error("Error: Could not load texture file " + file_name + " using stbi_load!");
-    }
+        spdlog::error("Could not load texture file {} using stbi_load!  Falling back to error texture.", file_name);
+
+        auto texture_data = load_error_texture_data(&m_texture_width, &m_texture_height, &m_texture_channels);
+        assert(!texture_data.empty());
 
-    spdlog::debug("Texture dimensions: width: {}, height: {}, channels: {}.", m_texture_width, m_texture_height,
-                  m_texture_channels);
+        // Calculate the memory size of the texture.
+        VkDeviceSize texture_memory_size = m_texture_channels * m_texture_width * m_texture_height;
 
-    // Calculate the memory size of the texture.
-    // We need 4 times the size since we have 4 channels: red, green, blue and alpha channel.
-    VkDeviceSize texture_memory_size = 4 * m_texture_width * m_texture_height;
+        create_texture(texture_data.data(), texture_memory_size);
+
+    } else {
+        // We are forcing 4 channels regardless of the disk representation of the image.
+        m_texture_channels = 4;
 
-    create_texture(texture_data, texture_memory_size);
+        spdlog::debug("Texture dimensions: width: {}, height: {}, channels: {}.", m_texture_width, m_texture_height,
+                      m_texture_channels);
 
-    // We can discard the texture data since we copied it already to GPU memory.
-    stbi_image_free(texture_data);
+        // Calculate the memory size of the texture.
+        VkDeviceSize texture_memory_size = m_texture_channels * m_texture_width * m_texture_height;
+
+        create_texture(texture_data, texture_memory_size);
+
+        // We can discard the texture data since we copied it already to GPU memory.
+        stbi_image_free(texture_data);
+    }
```

Comment:
```suggestion
    }
    stbi_image_free(texture_data);
```

Always free the texture data.

Reply:
*Move the comment too* ^^

Reply:
I updated this and condensed the commit history.

Reply:
You can resolve this conversation then 👍